src/main/resources/static/data-upload.js

```diff
@@ -149,5 +155,57 @@ function handle_error() {
 }
 
 $(document).ready(function () {
-    retrieveSessions();
+    $("#data-proceed-btn").attr("disabled","disabled");
+
+    $.ajax({
+        url: "/username",
+        type: "get",
+        success: function (data) {
+            USERNAME = data;
+        }
+    });
+
+    //for testing
+    $.ajax({
+        url: "/create-session",
+        type: "post",
+        data: {
+            name:"sessions1"
+        },
+        success: function (data) {
+            console.log('session1 created');
+            $.ajax({
+                url: "/create-session",
+                type: "post",
+                data: {
+                    name:"sessions2"
+                },
+                success: function (data) {
+                    console.log('session2 created');
+                    $.ajax({
+                        url: "/create-session",
+                        type: "post",
+                        data: {
+                            name:"sessions3"
+                        },
+                        success: function (data) {
+                            console.log('session3 created');
+                            retrieveSessions();
+                        },
+                        error: function () {
+                            console.log('Fail to load sessions.');
+                        }
+                    });
+                },
+                error: function () {
```

Comment:
Also in this case, defining error handler would increase readability. Especially that you define the same anonymous function three times. If you declared it as a variable and only referenced here it would be more readable and more concise.